src/components/UserSelector.tsx

```diff
@@ -12,24 +39,37 @@ export const UserSelector: React.FC = () => {
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={toggleList}
+          onBlur={hideList}
         >
-          <span>Choose a user</span>
+          {!currUser && <span>Choose a user</span>}
+          <span>{currUser?.name}</span>
```

Comment:
Replace the current line with conditional rendering to avoid displaying an empty span when there's no current user selected.